jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/EmailDeadlinesDBTest.java

```diff
@@ -0,0 +1,134 @@
+        props.setProperty(DBUserInfoImpl.EMAIL_QUERY, "select email from Users where entityId = ?");
+        props.setProperty(DBUserInfoImpl.LANG_QUERY, "select lang from Users where entityId = ?");
+        props.setProperty(DBUserInfoImpl.MEMBERS_QUERY, "select entityId from UserGroups where groupId = ?");
+
+        DBUserInfoImpl userInfo = new DBUserInfoImpl(props);
+                
+        this.taskService = (InternalTaskService) HumanTaskServiceFactory.newTaskServiceConfigurator()
+                                                .entityManagerFactory(emf)
+                                                .userInfo(userInfo)
+                                                .getTaskService();
+
+    }
+    
+    protected void prepareDb() {
+        try {
+            Connection conn = pds.getConnection();
+            String createUserTableSql = "create table Users (entityId varchar(255), email varchar(255), lang varchar(255), name varchar(255))";
+            PreparedStatement st = conn.prepareStatement(createUserTableSql);
+            st.execute();
+
+            String createGroupTableSql = "create table UserGroups (groupId varchar(255), entityId varchar(255), email varchar(255))";
+            st = conn.prepareStatement(createGroupTableSql);
+            st.execute();
+
+            // insert user rows
+            String insertUser = "insert into Users (entityId, email, lang, name) values (?, ?, ?, ?)";
+            st = conn.prepareStatement(insertUser);
+            
+            st.setString(1, "Darth Vader");
+            st.setString(2, "darth@domain.com");
+            st.setString(3, "en-UK");
+            st.setString(4, "Darth Vader");
+            st.execute();
+            
+            st = conn.prepareStatement(insertUser);
+            st.setString(1, "Tony Stark");
+            st.setString(2, "tony@domain.com");
+            st.setString(3, "en-UK");
+            st.setString(4, "Tony Stark");
+            st.execute();
+
+            st.close();
+            conn.close();
+
+        } catch (Exception e) {
+            e.printStackTrace();
+
+        }
+    }
+
+    @After
+    public void clean() {
+        TaskDeadlinesServiceImpl.reset();
+        super.tearDown();
+        cleanDb();
+        if (pds != null) {
+            pds.close();
+        }
+    }
+    
+    public void cleanDb() {
+        try {
+            Connection conn = pds.getConnection();
+            String dropUserTableSql = "drop table Users";
+            PreparedStatement st = conn.prepareStatement(dropUserTableSql);
+            st.execute();
+
+            String dropGroupTableSql = "drop table UserGroups";
+            st = conn.prepareStatement(dropGroupTableSql);
+
+            st.execute();
+
+            st.close();
+            conn.close();
+
+        } catch (Exception e) {
+            e.printStackTrace();
+
+        } finally {
+            pds.close();
```

Comment:
I would leave `pds` to be closed outside this method since it's being already closed at `clean` method. 

https://github.com/kiegroup/jbpm/blob/26921f4d212e561453888444861987322e453264/jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/EmailDeadlinesDBTest.java#L106

I would though move both `st` and `conn` variables closure to this `finally` block cause at the moment in case of any exception these won't be closed.

Reply:
I've refactored to use try-with-resources for taking advantage of autocloseable and creating auxiliary methods for enhancing readability. Thanks for your review @afalhambra !